### DISCUSSION TITLE
Empty slots forward as non-empty slots

### DISCUSSION THREAD

**Discussion by jtlapp** [sveltejs/kit#5767] [General]
I'm racing to meet a deadline and don't have time to create a minimal repro, but when you forward a slot to a child component and the slot happens to be empty, the child component `$$slots` flags it as non-empty.

I tried:

```
<slot slot="slotname" name="slotname" />
```

and

```
<svelte:fragment slot="slotname">{#if $$slots.slotname}<slot name="slotname" />{/if}{/svelte:fragment}
```

And the following doesn't work at all:

```
<slot slot="..." />
```

My svelte is getting pretty messy without this feature.

**Comment by rmunn** [comment]
This behavior is explicitly documented in [the Svelte docs for `$$slots`](https://svelte.dev/docs#template-syntax-slot-$$slots):

> Note that explicitly passing in an empty named slot will add that slot's name to `$$slots`. For example, if a parent passes `<div slot="title" />` to a child component, `$$slots.title` will be truthy within the child.

**Comment by jtlapp** [comment]
Okay thank you! It seems like this behavior makes slot forwarding incompatible with optional slots.
